src/Service/DatabaseLogger.php

```diff
@@ -6,21 +6,28 @@
 use Monolog\Logger;
 use Morebec\YDB\DatabaseConfig;
 use Morebec\YDB\Service\Database;
+use Morebec\YDB\Enum\LoggerChannel;
 
 /**
  * DatabaseLogger
  */
 class DatabaseLogger extends Logger
 {
+
+    private $loggers = [];
+
     /**
      * Constructs an instance of the database logger
      * @param string $databasePath path to the database
      */
     public function __construct(string $databasePath)
     {
-        parent::__construct('default');
-        $logsDir = $databasePath . "/" . Database::LOGS_DIR_NAME;
-        
-        $this->pushHandler(new RotatingFileHandler($logsDir . '/database.log', Logger::DEBUG));
+        $channels = LoggerChannel::getValues();
+        foreach($channels as $channel){
+            $this->loggers[$channel] = new Logger($channel);
+            $logsDir = $databasePath . "/" . Database::LOGS_DIR_NAME;
+            $this->pushHandler(new RotatingFileHandler($logsDir . "/".$channel."log", Logger::DEBUG));
```

Comment:
Also since it no longer relies on the Monolog Logger class's parent constructor, it should maybe drop the inheritance of the Monolog logger and instead implement the standard PHP Psr\LoggerInterface.
However, since the signature of the log method from that interface does not support the Channel Loggers, a new method should be added:
```php
DatabaseLogger::logWithChannel(string LogLevel, string $message, array $context = [], LoggerChannel $channel = null)
```